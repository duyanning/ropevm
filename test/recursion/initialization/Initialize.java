/* This regression test tries to deal with recursive class initialization, as stated in
   http://java.sun.com/docs/books/vmspec/2nd-edition/html/Concepts.doc.html#24237
   Unfortunately, the specification is very vague and there are no clear examples on
   all the possible cases. We try to reproduce some possible scenarions. Feel free to
   add those that you consider appropriate.
   So far, the results expected from this test are the same as those received by Sun's JDK
   and Kaffe. We did not base the test on what is stated in the specification but only on the
   output generated by these VMs.
*/

class Initialize
{
	public static void main(String[] args) {
		ClassA a = new ClassA();
		ClassB b = new ClassB();

		a.printInt();
		b.printInt();

		/*************************************/
		StaticInitA sta = new StaticInitA();
		StaticInitB stb = new StaticInitB();
	}
}

