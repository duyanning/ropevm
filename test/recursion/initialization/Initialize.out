6
4
Initializing A
Initializing B
Constructing A
Done initializing B
Constructing B
Done initializing A
Constructing A
Constructing B
